proposals/NNNN-unlock-existentials-with-fixed-associated-types.md

```diff
@@ -0,0 +1,249 @@
+# Unlock Existentials with Fixed Associated Types
+
+* Proposal: [SE-NNNN](NNNN-unlock-existentials-with-fixed-associated-types.md)
+* Authors: [Filip Sakel](https://github.com/filip-sakel), [Anthony Latsis](https://github.com/AnthonyLatsis)
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+
+## Introduction
+
+Protocols are a powerful feature of Swift. They are so diverse that they can be divided into two categories: regular protocols that can be used as types and [Protocols that are Unusable as Types](https://docs.swift.org/swift-book/LanguageGuide/Generics.html#ID189) (PUTs). PUTs are special because they reference 'Self' or contain associated type requirements making the fact they cannot be used as Types a sensible contraint. However, in some cases where a protocol inherits a PUT but also specifies all its associated types this constraint seems unintuitive. This proposal aims to relax this needless contraint allowing such protocols to be used as Types.
+
+## Motivation
+
+Currently any protocol conforming to a PUT becomes one itself as it inherits its associated types. However, some protocols specify these associated types, making this a frustrating limitations. The following example illustrates this problem:
+
+```swift
+protocol User: Identifiable 
+    where ID == String {
+    
+    var username: String { get }
+    var displayName: String { get }
+}
+
+extension User {
+    var id: String {
+        username
+    }
+}
+```
+
+Many would rightfully assume that `User` could be used as a Type, since we specify that the `ID` asscoated type inherited from [`Identifiable`](https://github.com/apple/swift-evolution/blob/master/proposals/0261-identifiable.md) is `String`. 
+
+```swift
+let myUser: User
+// ❌ `User` is a PUT
+```
+
+This is a great inconvenience with not so elegant workarounds:
+
+```swift
+protocol AnyUser { ... } 
+// No `Identifiable` conformance
+
+let myUser: AnyUser
+// ✅
+
+
+protocol User: AnyUser, Identifiable 
+    where ID == String { 
+    
+    var id: String {
+        username
+    }
+}
+
+let myOtherUser: User 
+// ❌ Still an error as expected
+```
```

Comment:
It might be worth talking about opaque result types as well, since you can avoid the error by writing “some User” but obviously you lose some type information.